modules/dnn/src/model.cpp

```diff
@@ -0,0 +1,265 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+// Copyright (C) 2018-2019, Intel Corporation, all rights reserved.
+// Third party copyrights are property of their respective owners.
+
+
+#include "precomp.hpp"
+#include <algorithm>
+#include <iostream>
+#include <utility>
+#include <iterator>
+
+#include <opencv2/imgproc.hpp>
+
+namespace cv {
+namespace dnn {
+
+Model::Model(const std::string& model, const std::string& config, int width,
+      int height, Scalar mean, float scale, bool swapRB, bool crop) : Net(readNet(model, config)),
+      width_(width), height_(height), mean_(mean), scale_(scale), swapRB_(swapRB), crop_(crop) {}
+
+void Model::setInputWidth(int width) {
+    width_ = width;
+}
+
+void Model::setInputHeight(int height) {
+    height_ = height;
+}
+
+void Model::setInputMean(Scalar mean) {
+    mean_ = mean;
+}
+
+void Model::setInputScale(float scale) {
+    scale_ = scale;
+}
+
+void Model::setInputCrop(bool crop) {
+    crop_ = crop;
+}
+
+void Model::setInputSwapRB(bool swapRB) {
+    swapRB_ = swapRB;
+}
+
+std::pair<int, float> Model::classify(const Mat& frame) {
+    Mat blob = blobFromImage(frame, scale_, Size(width_, height_), mean_, swapRB_, crop_);
+    setInput(blob);
+    Mat out = forward();
+    std::vector<float> result = out.reshape(1, 1);
+    std::vector<float>::iterator iter = std::max_element(result.begin(), result.end());
+
+    std::pair<int, float> prediction(iter - result.begin(), *iter);
+    return prediction;
+}
+
+void Model::detect(Mat& frame, std::vector<int>& classIds,
+                   std::vector<float>& confidences, std::vector<Rect2d>& boxes,
+                   float confThreshold, float nmsThreshold)
+{
+    Mat blob = blobFromImage(frame, scale_, Size(width_, height_), mean_, swapRB_, crop_);
+    setInput(blob);
```

Comment:
For Myriad target it's better to create uint8 blob and pass preprocessing parameters in `setInput`. So let's do the same in general case.